File: cs2.ts/.net.ts/system/io/file.ts
// @ts-nocheck
import { FileMode } from "./file-mode";
import { FileStream } from "./file-stream";
import { deleteFileBuffer, ensureFileBuffer, fileExists } from "./file-storage";

export class File {
    static exists(filePath: string): boolean {
        return fileExists(filePath);
    }

    static delete(filePath: string) {
        deleteFileBuffer(filePath);
    }

    static open(filePath: string, fileMode: FileMode): FileStream {
        ensureFileBuffer(filePath, fileMode);
        return new FileStream(filePath, fileMode);
    }

    static openRead(filePath: string): FileStream {
        ensureFileBuffer(filePath, FileMode.Open);
        return new FileStream(filePath, FileMode.Open);
    }

    static openWrite(filePath: string): FileStream {
        ensureFileBuffer(filePath, FileMode.OpenOrCreate);
        return new FileStream(filePath, FileMode.OpenOrCreate);
    }
}

File: cs2.cpp/.net.cpp/system/io/file.cpp
#include "file.hpp"
#include <fstream>

bool File::Exists(const char* fileName) {
	if (!fileName)
	{
		return false;
	}

	std::ifstream file(fileName);
	return file.good();
}

bool File::Delete(const char* fileName) {
	if (!fileName)
	{
		return false;
	}

	return std::remove(fileName) == 0;
}

FileStream File::Open(const char* filePath, FileMode fileMode)
{

}
